pkg/controller/utils/component.go

```diff
@@ -88,19 +87,14 @@ func (c componentHandler) CreateOrUpdateOrDelete(ctx context.Context, component
 	objsToCreate, objsToDelete := component.Objects()
 	osType := component.SupportedOSType()
 
-	for _, obj := range objsToCreate {
-		om, ok := obj.(metav1.ObjectMetaAccessor)
-		if !ok {
-			return fmt.Errorf("Object is not ObjectMetaAccessor")
-		}
-
+	createOrUpdateObject := func(obj client.Object, metaObj metav1.Object) error {
```

Comment:
Maybe this is just personal preference, but I don't really like have anonymous functions used like this. This makes the function more difficult to read because:

1. Variable scoping gets a little more confusing to read
2. You have a massive chunk of code that you basically need to skip over when reading the function because it's not called in line
3. Adds a lot of code that could just as easily be taken out to make the rest of the logic easier to follow (this is possibly the same point as 2)